### PR TITLE
Motor test

### PR DESCRIPTION
**Pull request by jaxxzer**
Test with Sub-3.5

### CODE REVIEW

File: src/QmlControls/APMSubMotorDisplay.qml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        subImage.source = getImage()
+    }
+
+    Image {
+        id: subImage
+        anchors.margins:    ScreenTools.defaultFontPixelWidth
+        anchors.top:        parent.top
+        anchors.bottom:     parent.bottom
+        anchors.left:       parent.left
+        anchors.right:      parent.right
```
Comment (patrickelectric) [inline src/QmlControls/APMSubMotorDisplay.qml]: fill parent ?

File: src/QmlControls/APMSubMotorDisplay.qml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+import QGroundControl.Palette       1.0
+import QGroundControl.ScreenTools   1.0
+import QGroundControl                       1.0
+import QGroundControl.FactSystem    1.0
+import QGroundControl.FactControls  1.0
+Item {
+    id: root
+
+    property var frameType
```
Comment (patrickelectric) [inline src/QmlControls/APMSubMotorDisplay.qml]: Maybe a default value ? Explicit is better than implicit.

File: src/QmlControls/APMSubMotorDisplay.qml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+                return "qrc:///qmlimages/Frames/Vectored.png"
+        case 2:
+                return "qrc:///qmlimages/Frames/Vectored6DOF.png"
+        case 4:
+                return "qrc:///qmlimages/Frames/SimpleROV-3.png"
+        case 5:
+                return "qrc:///qmlimages/Frames/SimpleROV-4.png"
+        }
+        return ""
+    }
```
Comment (patrickelectric) [inline src/QmlControls/APMSubMotorDisplay.qml]: function getImage() {
    var frameImagePath = "qrc:///qmlimages/Frames/"
    var frameImage = ""
    switch (frameType) {
    case 0:
            frameImage = "BlueROV1.png"
    case 1:
            frameImage = "Vectored.png"
    case 2:
            frameImage = "Vectored6DOF.png"
    case 4:
            frameImage = "SimpleROV-3.png"
    case 5:
            frameImage = "SimpleROV-4.png"
    }
    return frameImagePath + frameImage
}

File: src/AutoPilotPlugins/Common/MotorComponent.qml [hunk 4]
```diff
@@ -20,15 +20,26 @@ SetupPage {
     id:             motorPage
     pageComponent:  pageComponent
 
+    visibleWhileArmed: true
```
Comment (patrickelectric) [inline src/AutoPilotPlugins/Common/MotorComponent.qml]: unused ?

File: qgroundcontrol.qrc [hunk 5]
```diff
@@ -73,7 +74,7 @@
         <file alias="QGroundControl/Controls/MissionItemStatus.qml">src/PlanView/MissionItemStatus.qml</file>
         <file alias="QGroundControl/Controls/ModeSwitchDisplay.qml">src/QmlControls/ModeSwitchDisplay.qml</file>
         <file alias="QGroundControl/Controls/MultiRotorMotorDisplay.qml">src/QmlControls/MultiRotorMotorDisplay.qml</file>
-		<file alias="QGroundControl/Controls/NoMouseThroughRectangle.qml">src/QmlControls/NoMouseThroughRectangle.qml</file>
+        <file alias="QGroundControl/Controls/NoMouseThroughRectangle.qml">src/QmlControls/NoMouseThroughRectangle.qml</file>
```
Comment (jaxxzer) [inline qgroundcontrol.qrc]: huh? how did this get here

File: qgroundcontrol.qrc [hunk 6]
```diff
@@ -51,12 +51,13 @@
...
         <file alias="QGroundControl/Controls/AnalyzePage.qml">src/AnalyzeView/AnalyzePage.qml</file>
         <file alias="QGroundControl/Controls/AppMessages.qml">src/QmlControls/AppMessages.qml</file>
+        <file alias="QGroundControl/Controls/APMSubMotorDisplay.qml">src/QmlControls/APMSubMotorDisplay.qml</file>
         <file alias="QGroundControl/Controls/CameraSection.qml">src/PlanView/CameraSection.qml</file>
         <file alias="QGroundControl/Controls/ClickableColor.qml">src/QmlControls/ClickableColor.qml</file>
         <file alias="QGroundControl/Controls/DropButton.qml">src/QmlControls/DropButton.qml</file>
         <file alias="QGroundControl/Controls/ExclusiveGroupItem.qml">src/QmlControls/ExclusiveGroupItem.qml</file>
         <file alias="QGroundControl/Controls/FactSliderPanel.qml">src/QmlControls/FactSliderPanel.qml</file>
-		<file alias="QGroundControl/Controls/FileButton.qml">src/QmlControls/FileButton.qml</file>
+        <file alias="QGroundControl/Controls/FileButton.qml">src/QmlControls/FileButton.qml</file>
```
Comment (jaxxzer) [inline qgroundcontrol.qrc]: same

**Comment by jaxxzer** [comment]
@patrickelectric one last look? No longer wip

File: src/AutoPilotPlugins/Common/MotorComponent.qml [hunk 7]
```diff
@@ -97,42 +157,57 @@ SetupPage {
...
 
-                MultiRotorMotorDisplay {
+//                MultiRotorMotorDisplay {
+//                    anchors.top:    parent.top
+//                    anchors.bottom: parent.bottom
+//                    width:          height
+//                    motorCount:     controller.vehicle.motorCount
+//                    xConfig:        controller.vehicle.xConfigMotors
+//                    coaxial:        controller.vehicle.coaxialMotors
+//                }
```
[the remark this answers is not in the document]
Comment (jaxxzer) [reply src/AutoPilotPlugins/Common/MotorComponent.qml]: to remember to put it back :)

**Pull request merged**